lib/ansible/modules/get_url.py

```diff
@@ -506,12 +506,12 @@ def main():
             for line in lines:
                 parts = line.split(None, 1)
                 if len(parts) == 2:
-                    checksum_map[parts[0]] = parts[1]
+                    checksum_map[line] = (parts[0], parts[1])
```

Comment:
I don't mean to keep asking for changes, but the `line` is effectively meaningless for us in the end.  It seems a dict was the wrong choice initially.

Why not make `checksum_map` a `list` of `tuple` instead?

```
checksum_map = []
[...]
checksum_map.append((parts[0], parts[1]))
[...]
for cksum in (s for (s, f) in checksum_map if f.strip('./') == filename):
```